OverlayPlugin.Core/MemoryProcessors/Combatant/LineCombatant.cs

```diff
@@ -0,0 +1,411 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using RainbowMage.OverlayPlugin.MemoryProcessors.InCombat;
+using RainbowMage.OverlayPlugin.NetworkProcessors;
+
+namespace RainbowMage.OverlayPlugin.MemoryProcessors.Combatant
+{
+    public class LineCombatant
+    {
+        public const uint LogFileLineID = 261;
+        private ILogger logger;
+        private readonly FFXIVRepository ffxiv;
+        private ICombatantMemory combatantMemoryManager;
+        private IInCombatMemory inCombatMemory;
+        private ConcurrentDictionary<uint, CombatantStateInfo> combatantStateMap = new ConcurrentDictionary<uint, CombatantStateInfo>();
+
+        int offsetHeaderActorID;
+        int offsetHeaderLoginUserID;
+
+        // Only emit a log line when this information changes every X milliseconds
+        private struct CombatantChangeCriteria
+        {
+            // in milliseconds
+            public const int PollingRate = 20;
+
+            // in milliseconds
+            public const uint InCombatDelayDefault = 1000; // If any property has changed in this timeframe, a line will be written
+            public const uint OutOfCombatDelayDefault = 5000;
+
+            public const uint InCombatDelayNameID = 0;
+            public const uint OutOfCombatDelayNameID = 1000;
+
+            public const uint InCombatDelayPosition = 250;
+            public const uint OutOfCombatDelayPosition = 1250;
+
+            public const uint InCombatDelayTransformationID = 0;
+            public const uint OutOfCombatDelayTransformationID = 1000;
+
+            public const uint InCombatDelayWeaponID = 0;
+            public const uint OutOfCombatDelayWeaponID = 1000;
+
+            public const uint InCombatDelayTargetID = 0;
+            public const uint OutOfCombatDelayTargetID = 1000;
+
+            // in in-game distance
+            public const float InCombatDistancePosition = 5F;
+            public const float OutOfCombatDistancePosition = 15F;
+
+            // in radians
+            public const float InCombatDistanceHeading = (float)(45 * (Math.PI / 180)); // 45º turns
+            public const float OutOfCombatDistanceHeading = 20f; // Effectively disabled
+
+            // Check these fields for changes to determine if we should dump a full list of all changes
+            private static readonly string[] DefaultCheckFieldNames = new string[] {
+                "OwnerID",
+                "Type",
+                "MonsterType",
+                "Status",
+                "ModelStatus",
+                "AggressionStatus",
+                "IsTargetable",
+                "Name",
+                "Radius",
+                "BNpcID",
+                "CurrentMP",
+                "IsCasting1",
```

Comment:
That's what the field is called in `Combatant`:

https://github.com/OverlayPlugin/OverlayPlugin/blob/1160109e28cf361383d9d13526a3c2f78942dd53/OverlayPlugin.Core/MemoryProcessors/Combatant/Common.cs#L142

Reply:
Should we call it something different here or there? It seems awkward to propagate that out to log files.

Reply:
Then maybe it's worth changing this first before anybody depends on it? We've already got #172 as a backwards compat issue to fix.

Reply:
I'm not sure if anything downstream is already depending on this field, since it's included in the `getCombatants` response.

Reply:
It just seems like exposing internals to call something IsCasting1/IsCasting2 (and I certainly don't know what the difference is).

Maybe a deprecation path here is add a few aliased fields in combatants to handle `WeaponId`/`WeaponID` and `IsCasting1` so that we send duplicate info with different names for backwards compatibility, and then clean them up later in the future once people have had a warning and a chance to move off of them?  Alternatively we could `IsCasting1` and `IsCasting2` fields here (arguably 0x14 lines include this information already).

Reply:
The relevant FFXIVClientStructs struct is here, showing that `IsCasting1` => `IsCasting`, and `IsCasting2` => `Interruptible`:

https://github.com/aers/FFXIVClientStructs/blob/main/FFXIVClientStructs/FFXIV/Client/Game/Character/Character.cs#L103-L126